crypto/src/merkle/proofs.rs

```diff
@@ -4,9 +4,18 @@
 // LICENSE file in the root directory of this source tree.
 
 use crate::{errors::MerkleTreeError, Hasher};
-use std::collections::{BTreeMap, HashMap};
 use utils::{ByteReader, Deserializable, DeserializationError, Serializable};
 
+#[cfg(feature = "alloc")]
+use alloc::collections::BTreeMap;
+#[cfg(feature = "alloc")]
+use hashbrown::HashMap;
+#[cfg(feature = "std")]
+use std::collections::{BTreeMap, HashMap};
```

Comment:
If we have `utils::collections` module as described in the other comment, we could replace this with use `utils::collections::*`;